Include/cmsis_dsp_math_types.h

```diff
@@ -0,0 +1,18 @@
+/* Copyright (c) 2022 Google Inc
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_CMSIS_DSP_MATH_TYPES_H_
+#define INCLUDE_CMSIS_DSP_MATH_TYPES_H_
+
+#include "arm_math_types.h"
+
+#define CMSIS_MATH_SUCCESS ARM_MATH_SUCCESS
+#define CMSIS_MATH_ARGUMENT_ERROR ARM_MATH_ARGUMENT_ERROR
+#define CMSIS_MATH_LENGTH_ERROR ARM_MATH_LENGTH_ERROR
+#define CMSIS_MATH_SIZE_MISMATCH ARM_MATH_SIZE_MISMATH
+#define CMSIS_MATH_MANIF ARM_MATH_MANIF
+#define CMSIS_MATH_TEST_FAILURE ARM_MATH_TEST_FAILURE
+#define CMSIS_MATH_DECOMPOSITION_FAILURE ARM_MATH_DECOMPOSITION_FAILURE
```

Comment:
I may perhaps add those to `arm_math_types.h`. A developer can (for instance) only include `dsp/basic_math_functions.h` if there is no need to include all of the headers. For instance, if the application is only using a subset of the functions.

This `basic_math_functions.h` is including `arm_math_types.h` so the developer would only have access to the `ARM_MATH_` error codes without those defines.

There is no other public symbol that would need to be defined.

`basic_math_functions.h`  is including

```
#include "arm_math_types.h"
#include "arm_math_memory.h"

#include "dsp/none.h"
#include "dsp/utils.h"
```

The `arm_math_types.h` doesn't have any `arm_` prefix except for the error code.
The `arm_math_memory.h` is providing some memory functions with no `arm_` prefix.

`none.h` is used to provide C implementation of some intrinsics when they are missing. No `arm_` prefix.

`utils.h` is containing some functions with `arm_` prefix and so may also require a few define inside. For instance

`#define cmsis_dsp_div_int64_to_int32 arm_div_int64_to_int32`